api/apps/v1alpha1/nimbuild_types.go

```diff
@@ -0,0 +1,140 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// NIMBuildSpec to build optimized trtllm engines with given model config and weights.
+type NIMBuildSpec struct {
+	// Required: Reference to the model weights from NIMCache
+	NIMCacheName string `json:"nimCacheName"`
```

Comment:
We can make it as a NIMCacheRef to encapsulate both nimCache name and profile name.

Reply:
We can add env as input here